packages/autocertifier/src/DnsServer.ts

```diff
@@ -0,0 +1,250 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+
+import { DnsHandler, DnsResponse, Packet, createServer } from 'dns2'
+import { Database, Subdomain } from './Database'
+import { Logger } from '@streamr/utils'
+
+const logger = new Logger(module)
+
+// TODO: there is appears to be a general problem with typing in this class. Seems that the DNS2 library does
+// not provide proper typing for many fields and response types. Alternativel we should just simply send all
+// responses as the DnsResponse type instead of unknown
+export class DnsServer {
+
+    private server?: any
+    private readonly domainName: string
+    private readonly ownHostName: string
+    private readonly dnsServerPort: string
+    private readonly ownIpAddress: string
+    private readonly db: Database
+
+    constructor(domainName: string, ownHostName: string, dnsServerPort: string, ownIpAddress: string, db: Database) {
+        this.domainName = domainName
+        this.ownHostName = ownHostName
+        this.dnsServerPort = dnsServerPort
+        this.ownIpAddress = ownIpAddress
+        this.db = db
+    }
+
+    public async createSubdomain(subdomain: string, ipAddress: string, port: string, token: string): Promise<void> {
```

Comment:
What is the main functionality of this class? Maybe the DB operations should be outside of this class?